src/vmm/src/vmm_config/machine_config.rs

```diff
@@ -273,6 +273,18 @@ impl Default for CpuFeaturesTemplate {
     }
 }
 
+impl From<String> for CpuFeaturesTemplate {
+    fn from(c: String) -> Self {
+        match c.as_str() {
+            "C3" => CpuFeaturesTemplate::C3,
+            "T2" => CpuFeaturesTemplate::T2,
+            "T2S" => CpuFeaturesTemplate::T2S,
+            "None" => CpuFeaturesTemplate::None,
+            _ => CpuFeaturesTemplate::None,
+        }
+    }
+}
```

Comment:
- When a string is given which does not match we should not return `CpuFeaturesTemplate::None` but rather error.
- When a `From` implementation is fallible it is usualy better to go with a `TryFrom` implementation.
- Since the type doesn't contain the `String` and thus it isn't moved it would make more sense to implement it on `&str` then simply pass it like `CpuFeaturesTemplate::from(&my_string)`.

I would suggest:
```suggestion
#[derive(Debug,PartialEq,Eq)]
struct CpuFeaturesTemplateFromStringError;
impl std::fmt::Display for CpuFeaturesTemplateFromStringError {
    fn fmt(&self, f: &mut std::fmt::Formater<'_>) -> std::fmt::Result {
        write!(f,"Invalid CPU template string.")
    }
}
impl std::error::Error for CpuFeaturesTemplateFromStringError {}

impl TryFrom<&str> for CpuFeaturesTemplate {
    type Error = CpuFeaturesTemplateFromStringError;
    fn try_from(s: &str) -> Result<Self,Self::Error> {
        match s {
            "C3" => Ok(Self::C3),
            "T2" => Ok(Self::T2),
            "T2S" => Ok(Self::T2S),
            "None" => Ok(Self::None),
            _ => Err(CpuFeaturesTemplateFromStringError)
        }
    }
}
```